### PR TITLE
CI: enable unit tests for macos in github actions

### PR DESCRIPTION
**Pull request by mayankshah1607**
Signed-off-by: Mayank Shah <mayankshah1614@gmail.com>

**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
enable unit tests for macos in github actions

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/azurefile/nodeserver_test.go [hunk 1]
```diff
@@ -457,28 +465,30 @@ func TestNodeStageVolume(t *testing.T) {
...
-			Exec:      fakeExec,
-		}
-		d.cloud = &azure.Cloud{
-			Environment: azure2.Environment{StorageEndpointSuffix: "test_suffix"},
-		}
-
-		_, err := d.NodeStageVolume(context.Background(), &test.req)
-		if !testutil.AssertError(err, &test.expectedErr) {
-			t.Errorf("Desc: %s\nUnexpected error: %v\nExpected: %v", test.desc, err, test.expectedErr.GetExpectedError())
+		if !(test.skipOnDarwin && runtime.GOOS == "darwin") {
```
Comment (andyzhangx) [inline pkg/azurefile/nodeserver_test.go:468]: use `skip`?

Comment (andyzhangx) [reply pkg/azurefile/nodeserver_test.go:468]: continue

File: pkg/azurefile/nodeserver_test.go [hunk 2]
```diff
@@ -457,6 +465,9 @@ func TestNodeStageVolume(t *testing.T) {
 	d := NewFakeDriver()
 
 	for _, test := range tests {
+		if !(test.skipOnDarwin && runtime.GOOS == "darwin") {
+			continue
+		}
```
Comment (mayankshah1607) [inline pkg/azurefile/nodeserver_test.go:470]: @andyzhangx thanks, I have made the change

File: pkg/azurefile/nodeserver_test.go [hunk 3]
```diff
@@ -544,9 +558,11 @@ func TestNodeUnstageVolume(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		_, err := d.NodeUnstageVolume(context.Background(), &test.req)
-		if !testutil.AssertError(err, &test.expectedErr) {
-			t.Errorf("Desc: %v\nUnexcpected error: %v\nExpected: %v", test.desc, err, test.expectedErr.GetExpectedError())
+		if !(test.skipOnDarwin && runtime.GOOS == "darwin") {
```
Comment (andyzhangx) [inline pkg/azurefile/nodeserver_test.go:561]: use
```
		if test.skipOnDarwin && runtime.GOOS == "darwin" {
			continue
		}
```

File: pkg/azurefile/nodeserver_test.go [hunk 4]
```diff
@@ -262,9 +265,11 @@ func TestNodeUnpublishVolume(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		_, err := d.NodeUnpublishVolume(context.Background(), &test.req)
-		if !testutil.AssertError(err, &test.expectedErr) {
-			t.Errorf("Desc: %s\nUnexpected error: %v\nExpected: %v", test.desc, err, test.expectedErr.GetExpectedError())
+		if !(test.skipOnDarwin && runtime.GOOS == "darwin") {
```
Comment (andyzhangx) [inline pkg/azurefile/nodeserver_test.go:268]: use 
```
		if test.skipOnDarwin && runtime.GOOS == "darwin" {
			continue
		}
```

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve
/retest

**Pull request merged**